src/main/java/com/blog/web/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,38 @@
+package com.blog.web.security;
+
+import com.blog.web.models.UserEntity;
+import com.blog.web.repository.UserRepository;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.stream.Collectors;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private UserRepository userRepository;
+
+    public CustomUserDetailsService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        UserEntity user = userRepository.findFirstByUsername(username);
+        if(user != null) {
+            User authUser = new User(
```

Comment:
looks like UserEntity is being used for making a user -> can just have a constructor that takes a user

Reply:
User is an existing class provided by org.springframework.security.core.userdetails which we use to do authentication, we convert the UserEntity object into that. I don't believe I can add a constructor to the User class unless I extend it. Unless I am misunderstanding something.

I encapsulated it into a method in the UserEntity class which converts and returns the User object. Is that good?